isis/tests/Hyabusa2OncCameraTests.cpp

```diff
@@ -0,0 +1,245 @@
+#include <QDebug>
+
+#include <iomanip>
+#include <iostream>
+
+#include "Hyb2OncCamera.h"
+#include "Camera.h"
+#include "CameraFactory.h"
+#include "IException.h"
+#include "iTime.h"
+#include "Preference.h"
+#include "Pvl.h"
+#include "TestUtilities.h"
+#include "PvlGroup.h"
+#include "Fixtures.h"
+
+#include "gmock/gmock.h"
+
+using namespace std;
+using namespace Isis;
+
+void testCamera(Cube &c, double knownLat, double knownLon,
+                double s1, double l1, 
+                double s2, double l2, 
+                double s3, double l3, 
+                double s4, double l4);
+void testLineSamp(Camera *cam, double sample, double line);
+
+TEST_F(DefaultCube, Hayabusa2OncCameraW1CameraTest) {
+  PvlGroup &kernels = testCube->label()->findObject("IsisCube").findGroup("Kernels");
+  kernels.findKeyword("NaifFrameCode").setValue("-37110");    
+  
+  PvlGroup &inst = testCube->label()->findObject("IsisCube").findGroup("Instrument");
+  inst.findKeyword("InstrumentId").setValue("ONC-W2");
+  inst.findKeyword("SpacecraftName").setValue("HAYABUSA-2");
+  PvlKeyword startcc("SpacecraftClockStartCount", "33322515");
+  PvlKeyword stopcc("SpaceCraftClockStopCount", "33322515");
+  PvlKeyword binning("Binning", "1");
+  inst += startcc;
+  inst += stopcc; 
+  inst += binning; 
+
+  PvlObject &naifKeywords = testCube->label()->findObject("NaifKeywords");
+  
+  json nk; 
+  nk["INS-37110_FOCAL_LENGTH"] = 10.44;
+  nk["INS-37110_PIXEL_PITCH"] = 0.013;
+  nk["INS-37110_TRANSX"] = {0.0, 0.013, 0.0};
+  nk["INS-37110_TRANSY"] = {0.0, 0.0, 0.013};
+  nk["INS-37110_ITRANSS"] = {0.0, 76.923076923077, 0.0};
+  nk["INS-37110_ITRANSL"] = {0.0, 0.0, 76.923076923077};
+  nk["INS-37110_BORESIGHT_LINE"] = 490.5;
+  nk["INS-37110_BORESIGHT_SAMPLE"] = 512.5;
+  nk["INS-37110_OD_K"] = {1.014, 2.933e-07, -1.384e-13};
+  nk["BODY499_RADII"] = {3396.19, 3396.19, 3376.2};
+  nk["CLOCK_ET-37_33322515_COMPUTED"] = "8ed6ae8930f3bd41";
+  nk["BODY_CODE"] = 499;
+  nk["BODY_FRAME_CODE"] = 10014; 
+  PvlObject newNaifKeywords("NaifKeywords", nk);
+  naifKeywords = newNaifKeywords; 
```

Comment:
In a perfect world, we would be able to instantiate an arbitrary camera model with some test ephem data (e.g. looking at some non real target and simple pointing.position data etc.) without having to more or less go through the spice init process.  But since this isn't an option (I didn't notice anything in the ISIS API that would allow it), wouldn't these issues be caught in the tests? I specifically had to go through the camera model to see what the model needed while using a real cube Hayabusa-2 image for reference.

Reply:
> I specifically had to go through the camera model to see what the model needed while using a real cube Hayabusa-2 image for reference.

This is the part I am concerned about. If the person writing the test messes up here, it can slip through. It's unlikely but we've had issues with things like this before.

Reply:
Are you suggesting we just make it go through the spiceinit process? More specifically, make a fixture using real cubes and ephemeris data per instrument? I'm not sure of an alternative.

Reply:
I'm not worried about the table data and stuff that gets attached during the spiceinit process. I'm worried about the stuff that gets propagated to the label during ingestion. All of the values in the Instrument group mainly.

Reply:
Here's the ONC-T Instrument group from the ingestion app test. Everything in this group could be used by the camera model. It seems like we're using bad test data if the instrument group of the test cube only contains a few of these values *and* contains values from a completely different image/instrument. It's a question of how "black box" should the testing on this be. This test is written based entirely off the code, not the expected use.

```
  Group = Instrument
    SpacecraftName                  = HAYABUSA-2
    InstrumentId                    = ONC-T
    InstrumentName                  = "Optical Navigation Camera"
    OriginalTargetName              = Earth
    TargetName                      = Earth
    StartTime                       = 2015-12-04T04:09:08.547
    StopTime                        = 2015-12-04T04:09:08.555
    ExposureDuration                = 0.0082 <seconds>
    SpacecraftClockStartCount       = 1/1012716101
    SpacecraftClockStartCount       = 1/1012716133
    RawSpacecraftClockCount         = 0x3C5CD245 <1/32 sec>
    OnboardDataRecordedTime         = 3C5CE063
    Binning                         = 1
    SelectedImageAreaX1             = 1
    SelectedImageAreaY1             = 1
    SelectedImageAreaX2             = 1024
    SelectedImageAreaY2             = 1
    SelectedImageAreaX3             = 1
    SelectedImageAreaY3             = 1024
    SelectedImageAreaX4             = 1024
    SelectedImageAreaY4             = 1024
    SmearCorrection                 = ONBOARD
    OffsetCorrection                = N/A
    FlatCorrection                  = NON
    RadianceConversion              = NON
    PhotometricCorrection           = NON
    BandRegistration                = NON
    L2BFlatFileName                 = N/A
    L2BSystemEfficiencyFileName     = N/A
    L2CShapeModelFileName           = N/A
    L2DPhaseFunctionFileName        = N/A
    L2DShapeModelFileName           = N/A
    ImageID                         = 36086
    SubImageCount                   = 2
    BusLineVoltage                  = 49.28 <V>
    ONCCurrent                      = 0.51 <V>
    FLACCurrent                     = 0.00 <V>
    ONCAETemperature                = 0.37 <degC>
    ONCTOpticsTemperature           = 19.21 <degC>
    ONCTCCDTemperature              = -29.46 <degC>
    ONCTElectricCircuitTemperature  = -10.31 <degC>
    ONCW1OpticsTemperature          = 1.47 <degC>
    ONCW1CCDTemperature             = -25.23 <degC>
    ONCW1ElectricCircuitTemperature = -10.29 <degC>
    ONCW2OpticsTemperature          = 1.11 <degC>
    ONCW2CCDTemperature             = -25.04 <degC>
    ONCW2ElectricCircuitTemperature = -6.16 <degC>
    FLACTemperature                 = -13.40 <degC>
    Compression                     = LOSSLESS
  End_Group
```